### ISSUE TITLE
[Bug]: MO crash running a query

### ISSUE THREAD

**Issue by dengn** [matrixorigin/matrixone#4537]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93): 0.5.0
- Hardware parameters: macbook m1 4c16g arm
- OS type: osx
- Others:
```

### Actual Behavior

For a table with the following data structure and about 11 millions rows:

CREATE TABLE pe (
    -> ts_code VARCHAR(255) DEFAULT NULL,
    -> trade_date VARCHAR(255) DEFAULT NULL,
    -> pe FLOAT DEFAULT NULL,
    -> pb FLOAT DEFAULT NULL
    -> );

mysql> select * from pe limit 10;
+-----------+------------+----------+--------+
| ts_code   | trade_date | pe       | pb     |
+-----------+------------+----------+--------+
| 300045.SZ | 20170911   | 128.1646 | 5.0357 |
| 300045.SZ | 20170908   | 128.5033 |  5.049 |
| 300045.SZ | 20170907   | 131.2134 | 5.1555 |
| 300045.SZ | 20170906   |  133.246 | 5.2354 |
| 300045.SZ | 20170905   | 128.8421 | 5.0623 |
| 300045.SZ | 20170904   | 129.5196 |  5.089 |
| 300045.SZ | 20170901   | 126.3578 | 4.9647 |
| 300045.SZ | 20170831   | 126.4708 | 4.9692 |
| 300045.SZ | 20170830   | 125.2286 | 4.9204 |
| 300045.SZ | 20170829   | 124.4382 | 4.8893 |
+-----------+------------+----------+--------+

Run query as:

mysql> SELECT pe1.ts_code, pe1.pe, pe1.pb
    -> FROM pe as pe1
    -> WHERE pe1.pe = (SELECT min(pe2.pe)
    ->             FROM pe as pe2
    ->          WHERE pe1.ts_code = pe2.ts_code)
    -> ORDER BY trade_date
    -> DESC LIMIT 1;
ERROR 2013 (HY000): Lost connection to MySQL server during query

MO crashed, with a super long error message, I can only catch part of it.
[error.log](https://github.com/matrixorigin/matrixone/files/9355606/error.log)

**Comment by LeftHandCold** [comment]
<img width="1279" alt="image" src="https://user-images.githubusercontent.com/14086886/185089097-43577b79-4316-4a62-9346-10ad9a63034d.png">
0.5.0 mo-server query will be much slower than 0.6

**Comment by dengn** [comment]
Fixed in 0.6
